### ISSUE TITLE
Show task in Calendar

### ISSUE THREAD

**Issue by arkhan** [xsoh/Etar-Calendar#83]

**Comment by jspricke** [comment]
Etar only displays what's available in the in the Android database. Which tasks do you mean?

**Comment by arkhan** [comment]
I have Baikal Server, un my phone DAVdroid + opentask, but Etar only show events, no show tasks rugiste red in my webdav calendar

**Comment by terba** [comment]
I would also use this feature. A task with a deadline or start date should be visible in the calendar (as is in any calendaring solution I've ever seen). But I can imagine what Google did with that too...

**Comment by jspricke** [comment]
opentask would need to export them into the calendar database, or we would need to add support for opentask. I would prefer the first version, as it would be more general.

(Personal comment: I have a server side script do do that, but but I use a special server as well).

**Comment by davidhedlund** [comment]
[mylabel:important](https://github.com/search?utf8=%E2%9C%93&q=commenter%3Adavidhedlund+%22mylabel%3Aimportant%22+in%3Acomments&type=Issues&ref=searchresults)

**Comment by smichel17** [comment]
This is related to a feature I wish calendar apps had: adding standalone reminders (ie, not tied to an event). You can approximate this with 0 length events but it's a poor substitute.

**Comment by ramack** [comment]
It seems to be better to put this functionality into the task app and have a calendar provider integrated there. (e. g. in [OpenTasks](https://github.com/dmfs/opentasks/issues/603).
Therefore I propose to close this issue in Etar...

**Comment by dilyanpalauzov** [comment]
I do not get how this is supposed to work.  CalDAV has the VEVENT and VTODO components.  Per definition, events occupy time in the calendar, but tasks take no time on the calendar.

A VTODO can contain Start, Completed and Due date.

Should tasks be displayed in the calender when any of those properties are provided?

**Comment by terba** [comment]
KOrganizer, what I use on desktop, displays only tasks with due date in the calendar views. They are shown at the due date/time of course. Start and Completed make no sense in this, so if a task has no due date, it will not be displayed at all.

**Comment by sylveryte** [comment]
[TodoAgenda](https://github.com/andstatus/todoagenda) shows both tasks and calendar events, from android database(i think), the tasks are added using [OpenTasks](https://github.com/dmfs/opentasks) and synced to nextcloud server using caldav.

Could this be helpful? I also want a calendar app with tasks and calendar events.

**Comment by dcorking** [comment]
Does Android Calendar Sync API support the "Reminders" that can be made in Google Calendar (web app and android app?

![image](https://user-images.githubusercontent.com/2813357/74658617-97108380-518a-11ea-9aef-8027c7b428e8.png)

https://calendar.google.com/calendar/r

If so then I can replace Google Calendar mobile app https://play.google.com/store/apps/details?id=com.google.android.calendar with Etar. If not, then tasks/reminders are probably out of scope of Etar.

**Comment by AlexandreBonneau** [comment]
> KOrganizer, what I use on desktop, displays only tasks with due date in the calendar views. They are shown at the due date/time of course. Start and Completed make no sense in this, so if a task has no due date, it will not be displayed at all.

Exactly.
For instance, I've uploaded a picture of what tasks looks like in Korganizer in that [issue](https://github.com/Etar-Group/Etar-Calendar/issues/491#issuecomment-591265979).

Tasks (VTODO) should be shown on due date/time (and could take up a 15 minutes space so that you could read the tasks description a bit).
Since there can be much more tasks than events in a given day, perhaps give the user the possibility to display tasks depending of the view type?
For instance I would gladly display my tasks in the week, day and agenda views, but not in the month view.

**Comment by Senior-Discount** [comment]
I just discovered this. I just setup a NextCloud + DAVx5 + Etar + Tasks.org configuration and noticed that:
-  NextCloud Calendar shows both events and tasks. 
- Etar only displays events.
- Tasks only display tasks

Since NextCloud can display both events and tasks (I haven't tried their mobile app yet, only web), I'm sure it could be possible for Etar, however I don't know what complexity it will bring. Also since I'm able to have my events and tasks synced through my current configuration; having tasks and events separate in their own apps is fine however I wonder if it's possible to implement this feature.

**Comment by AlexandreBonneau** [comment]
@Senior-Discount I'm using almost the same setup with `NextCloud`+ `DAVx5` + `Etar` + `OpenTasks`.
However while `OpenTasks` is very good to list and track your tasks, it fails at displaying a global view (Day/Week) to organize them all (but OpenTasks is not designed for this, Etar should be).

That view is pretty important when you have to make an appointment ; currently with Etar you can see 'free' spaces, however you have to switch to another app (`OpenTasks`) to check if you have any tasks in those free spaces.
This is not really handy, hence the feature request :>

**Comment by wolfisberg** [comment]
I would love that feature as well.

I just noticed that both web / desktop calendars I use (thunderbird and nextcloud) display tasks with due dates in the calendar, while all android calendar apps I tested (google, simple calendar, etar) can't seem to display tasks. I believe to remember to have read somewhere that the task list thing is an extension to the CalDev standrad, so maybe that's the missing piece?

I was using todoist before, and managed to import those tasks into the google calendar app (maybe todoist converts them to "pure" CalDev in order to provide them...).

All in all I perceive this feature as pretty essential for anybody that uses tasks since having to check tasks and events in different apps is not only annoying, but also the cause of mis-planing or even missed events/deadlines if not diligently done.

**Comment by lemexicainendormi** [comment]
Tasks.org android app can create calendar event when adding new task. Not ideal because if you have many calendars you have to select it manually for each task and also you get duplicate entries in other clients (thunderbird, nextcloud). But that's a workaround for me now. Nice thing is it updates the created event name with completed when you mark the task complete.

**Comment by tamas646** [comment]
Just to mention: There's a pull request which adds OpenTasks support: #1279

**Comment by 31337-4554551n** [comment]
I know this is an 8 year old request, but I'm adding my voice to the pile.
Would love to see my calendar events on one pane of glass in etar like I can in the nextcloud calendar webui

**Comment by 9999999design** [comment]
tldr skitzo slop: **todo agenda** apk widget only

Anyone else try Etar, Business Calendar 2, Calengoo, Rainmeter, Thunderbird, Betterbird (after thunderbird crashes) and then start reading git glitch issues on thunderbird from 2004 that have not yet been solved but people write to the request after 20+ years. Then you switch to paper and pen and then put a pen in your pocket and then end up with the pen jammed into your palm and bleed all over your keys and pants too? And then try voice integration with home assistant and then lose the ability to use the microphone because you're on call and android took away the ability to use the microphone during the app use and then you start to make your own calendar app and realize you got nothing done because you were so busy organzing the organizer that glitched out after installing dav5x, ics5x, baikal, raciale on a thinkpad with TLP battery limitations to prevent fire risk on a docker image with firewall rules that has access through tailscale then you switch to wireguard and open a UDP port for you IP that changes, so you need to get a DDNS for your DHCP IP LAN? 

Then you switch back to google calendar? And then you get fed up with Google and switch to a Chinese phone and then you don't know what to do anymore?

Jokes aside (very few) I use ToDo agenda widget, it shows me whats coming up and what happened earlier (up to your own setting) and it keeps overdues at the top. It's no real calendar view to take in via the weekly, daily, view. But I've tried just about every app. Even the business calendar 2 pro thing claims on Google Play to have Dav5x sync integration with calendar and tasks, it does not have tasks integration with dav5x. 

The creators of DAV have written their own book about the creation of it, not interested in reading it, but it's an old abandoned standard that has been re-rewritten by Apple, Google, TickTick, ect. 

I've tried making my own in Python for terminal and it's a nightmare dealing with the varying documented VTODO stuff that may or may not have % progress complete areas, notes, ect. 


ToDo Agenda's F-Droid APK is great, but I long for something like thunderbird's full desktop view of what's coming up this week and what's overdue and what's happening today on one solid screen. Then I can write a simple schedule for myself on my phone homescreen.

Calendar integration is a mess between vendors that are writing spaghetti that will end up with .ics import errors for timezones and other crap if you have to deal with Google Calendar > other calendar with timezone codes between countries for work, I've almost missed intercontinental work meetings as a result of this mess.

Maybe one day we get a dav5x integrated calendar with tasks and calendar events. We only have thunderbird for desktop and todo agenda widget for android, nothing else is stable, usable, workable, stable-ish.
